wallets/keplr-extension/src/extension/main-wallet.ts

```diff
@@ -18,7 +18,10 @@ export class KeplrExtensionWallet extends MainWalletBase {
     this.initingClient();
     try {
       const keplr = await Keplr.getKeplr();
-      this.initClientDone(keplr ? new KeplrClient(keplr) : undefined);
+      if (!keplr) {
+        throw ClientNotExistError;
```

Comment:
I'm not sure if we're supposed to use `this.initClientError` instead of directly throwing?

I think teams like Osmosis have a option to avoid errors being thrown, which is why all wallets have done this:

https://github.com/cosmology-tech/cosmos-kit/blob/main/wallets/fin-extension/src/extension/main-wallet.ts
https://github.com/cosmology-tech/cosmos-kit/blob/main/wallets/galaxy-station-extension/src/extension/main-wallet.ts
https://github.com/cosmology-tech/cosmos-kit/blob/main/wallets/imtoken-extension/src/extension/main-wallet.ts
https://github.com/cosmology-tech/cosmos-kit/blob/main/wallets/leap-extension/src/extension/main-wallet.ts
https://github.com/cosmology-tech/cosmos-kit/blob/main/wallets/ninji-extension/src/extension/main-wallet.ts
https://github.com/cosmology-tech/cosmos-kit/blob/main/wallets/coin98-extension/src/extension/main-wallet.ts